src/parser/mod.rs

```diff
@@ -3550,6 +3550,17 @@ impl<'a> Parser<'a> {
             None
         };
 
+        let autoincrement_offset = if self.parse_keyword(Keyword::AUTO_INCREMENT) {
```

Comment:
https://dev.mysql.com/doc/refman/8.0/en/create-table.html

I got super confused for a moment because `AUTO_INCREMENT` can occur either as part of the table options (as this PR correctly does)

```

table_options:
    table_option [[,] table_option] ...

table_option: {
    AUTOEXTEND_SIZE [=] value
  | AUTO_INCREMENT [=] value
  | AVG_ROW_LENGTH [=] value
...
```

Or also as part of the column definition

```

column_definition: {
    data_type [NOT NULL | NULL] [DEFAULT {literal | (expr)} ]
      [VISIBLE | INVISIBLE]
      [AUTO_INCREMENT] [UNIQUE [KEY]] [[PRIMARY] KEY]
...
```



Which SQLParser already handles
https://github.com/sqlparser-rs/sqlparser-rs/blob/41e47cc0136c705a3335b1504d50ae8b22711b86/src/ast/ddl.rs#L587